### DISCUSSION TITLE
Object Browser filter forcing me to refresh

### DISCUSSION THREAD

**Discussion by Gregory1701** [codefori/vscode-ibmi#1304] [IBM i]
Hi guys,
Not sure why this is happening, but I don't think it used to happen and I talked with a coworker and it's working as I think it should for him.  When I open VS Code, go to the IBM i extension, connect to the system I regularly work on, it brings up a list of my object browser filters; all collapsed.  When I click to expand my filter which I use daily it acts like there's nothing in it, but after I click the refresh button it loads.
Here's my issue in action:

https://github.com/halcyon-tech/vscode-ibmi/assets/132368707/30adcc9b-6439-4e3c-9b84-d94e31574cf3

Wondering if something was corrupt with my filter, I created a new one.  Same behavior.  I wondered if something was messed up with my connection, so I created a new connection to the same system and a new filter there; same behavior.  Any ideas?
[VS Code Developer Tools 5252023.txt](https://github.com/halcyon-tech/vscode-ibmi/files/11564502/VS.Code.Developer.Tools.5252023.txt)

**Comment by worksofliam** [comment]
@Gregory1701 What version of VS Code is this?

**Reply by Gregory1701** [reply]
![image](https://github.com/halcyon-tech/vscode-ibmi/assets/132368707/52f0e88d-c018-463b-9c6e-f08f5a5afe7e)

**Reply by worksofliam** [reply]
I cannot recreate on my Macbook. @chrjorgensen Any chance you can recreate this on Windows? Or maybe I need to boot my Windows system up.

**Reply by Gregory1701** [reply]
Interestingly, I just tried the same filter expansion on my IFS browser filter and it behaves the same - I have to click refresh for it to show me my contents.

**Reply by worksofliam** [reply]
My initial thoughts is that it might have to do with caching, but the IFS Browser doesn't have a cache. Could be a deeper VS Code issue

**Reply by worksofliam** [reply]
Please share the contents of your Code for IBM i output channel after the list doesn't appear before you click refresh.

<img width="939" alt="image" src="https://user-images.githubusercontent.com/3708366/178739532-03c8ad30-84de-4698-9904-65938f2ef651.png">

**Reply by worksofliam** [reply]
From your video, it doesn't even look like the request to the server is happening when you expand your TreeItem, but only when the refresh happens.

**Reply by chrjorgensen** [reply]
Please check [this issue](https://github.com/halcyon-tech/vscode-ibmi/issues/1221) - I think it has been resolved, but not yet in any release...

**Reply by worksofliam** [reply]
@chrjorgensen Thanks for pointing that out. We are days away from release. Hopefully that will solve it!

**Reply by Gregory1701** [reply]
Cool, I'm even more excited for the next release then!
